packages/template-block/src/components/TemplateText.tsx

```diff
@@ -46,49 +66,46 @@ export const TemplateText = ({
     const memoTitleRte = useMemo(
         () => (
             <RichTextEditor
-                id="template-block-title"
-                key={key}
+                id={`title-${blockId}`}
                 value={title ?? convertToRteValue(TextStyles.heading3)}
                 placeholder="Add a title"
-                onTextChange={setTitle}
+                onTextChange={saveTitle}
                 isEditing={isEditing}
                 showSerializedText={hasRichTextValue(title)}
                 plugins={customTitlePlugins}
+                key={titleKey}
```

Comment:
```suggestion
                key={templateKey}
```

use here the passed one